src/java/org/apache/cassandra/cql3/constraints/ColumnConstraints.java

```diff
@@ -120,18 +120,21 @@ public boolean hasRelevantConstraints()
     public void validate(ColumnMetadata columnMetadata) throws InvalidConstraintDefinitionException
     {
         if (!columnMetadata.type.isConstrainable())
+        {
             throw new InvalidConstraintDefinitionException("Constraint cannot be defined on the column "
                                                            + columnMetadata.name + " of type " + columnMetadata.type.asCQL3Type()
-                                                           + " for the table " + columnMetadata.ksName + "." + columnMetadata.cfName);
+                                                           + " for the table " + columnMetadata.ksName + '.' + columnMetadata.cfName + '.' +
+                                                           (columnMetadata.type.isCollection() ? " When using collections, constraints can be used only of frozen collections." : ""));
+        }
+
+        // this validation will check whether it makes sense to execute such constraint on a given column
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.validate(columnMetadata);
```

Comment:
Why checking the actual constraints before the satisfiability check? I think this may be potentially more expensive (thinking about the regex constraint)